### PR TITLE
add general context for item and basket

### PR DESCRIPTION
**Pull request by TG-hub**
created basic context for basket and item.

### CODE REVIEW

File: src/context/basket.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+export const BasketContext = createContext({
+    basket: '',
+    updateBasket: () => null
+})
+
+const BasketProvider = ({children}) => {
+    [basket, setBasket] = useState ('')
+
+    return (
+        <BasketContext.Provider value={{basket, setBasket: updateBasket}}>
```
Comment (JohnAGill) [inline src/context/basket.js]: setBasket and updateBasket are backwards here

File: src/context/items.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+export const itemContext = createContext({
+    item: '',
+    updateItem: () => null
+})
+
+const ItemProvider = ({children}) => {
+    const [item, setItem] = useState ('')
+
+    return(
+        <itemContext.provider value={{item, setItem: updateItem}}>
```
Comment (JohnAGill) [inline src/context/items.js]: setItem and updateItem are backwards here

File: src/App.js [hunk 3]
```diff
@@ -9,6 +9,7 @@ import SignUp from './pages/sign-up'
 import Confirmation from './pages/confirmation'
 import Item from './pages/item'
 import Checkout from './pages/checkout'
+import ContextSource from './context/root'
```
Comment (JohnAGill) [inline src/App.js:12]: You've imported ContextSource but you haven't used it

Comment (TG-hub) [reply src/App.js:12]: wrapped the whole App return() in a <ContextProvider> tag so hopefully that's fixed.

File: src/context/items.js (hunk 2 above)
Comment (TG-hub) [reply src/context/items.js]: also swapped these!

File: src/context/basket.js (hunk 1 above)
Comment (TG-hub) [reply src/context/basket.js]: swapped them around so should work.

**Review by JohnAGill — APPROVED** [review]

**Pull request merged**